### PR TITLE
doc draft for collections

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

collections

Document now available at https://docs.ansible.com/ansible/devel/collections_tech_preview.html

### CODE REVIEW

File: docs/docsite/rst/collections_tech_preview.rst [hunk 1]
```diff
@@ -0,0 +1,313 @@
+.. _collections:
+
+Collections
+===========
+
+Collections are a distribution format for Ansible content. They can be used to
+package and distribute playbooks, roles, modules, and plugins.
+You will be able to publish and use collections via `Ansible's Galaxy repository <http://galaxy.ansible.com>`_.
+
+.. note::
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
.. important::
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+    This is a Tech-Preview feature and is only supported by Ansible 2.8 (or greater).
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
    This feature is available in Ansible 2.8 as a *Technology Preview* and therefore is not fully supported. It should only be used for testing  and should not be deployed in a production environment.
```

**Review by samccann — COMMENTED** [review]
Adding a techpreview as important note.

**Review by samccann — COMMENTED** [review]
In order to remove that toc-tree error, we need to either have :orphan: at the top of this file or put it in the toc for some other guide. My nickel is to put it here:
https://docs.ansible.com/ansible/devel/reference_appendices/galaxy.html

But let's see what @acozine thinks...

File: docs/docsite/rst/collections_tech_preview.rst [hunk 3]
```diff
@@ -0,0 +1,313 @@
+.. _collections:
+
+Collections
+===========
+
+Collections are a distribution format for Ansible content. They can be used to
+package and distribute playbooks, roles, modules, and plugins.
+You will be able to publish and use collections via `Ansible's Galaxy repository <http://galaxy.ansible.com>`_.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
You will be able to publish and use collections through `Ansible's Galaxy repository <http://galaxy.ansible.com>`_.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+
+.. important::
+    This feature is available in Ansible 2.8 as a *Technology Preview* and therefore is not fully supported. It should only be used for testing  and should not be deployed in a production environment.
+    Future Galaxy or Ansible releases may introduce breaking changes.
+
+
+Collection Structure
+====================
+
+Collections follow a simple data stucture, none of the directories are required unless you have specific content that belongs in one of them. They do require a ``galaxy.yml`` at the root level of the collection. This file contains all of the metadata that Galaxy
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
Collections follow a simple data stucture. None of the directories are required unless you have specific content that belongs in one of them. They do require a ``galaxy.yml`` file at the root level of the collection. This file contains all of the metadata that Galaxy
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+and other tools need in order to package, build and publish it.::
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
and other tools need in order to package, build and publish the collection.::
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+    │   ├── vars/
+    │   ├── templates/
+    │   └── tasks/
+    └── tests/
+
+
+.. note::
+    * We will only accept ``.yml`` extensions for galaxy.yml.
+    * A full structure can be found at `Draft collection <https://github.com/bcoca/collection>`_
+    * Not all directories are currently in use, those are placeholders for future features
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
    * Not all directories are currently in use. Those are placeholders for future features.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+
+
+galaxy.yml
+----------
+
+This file contains the information about a colleciton neccessary for Ansible tools to operate.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
This file contains the information about a collection that is necessary for Ansible tools to operate.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+``galaxy.yml`` has the following fields (subject to expansion):
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
``galaxy.yml`` has the following fields (subject to changes and expansion):
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+        "other_namespace.collection1": ">=1.0.0"
+        "other_namespace.collection2": ">=2.0.0,<3.0.0"
+        "anderson55.my_collection": "*"    # note: "*" selects the highest version available
+    license:
+        - "MIT"
+    tags:
+        - demo
+        - collection
+    repository: "https://www.github.com/my_org/my_collection"
+
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst:78]: don't we also have other optional fields for docsite and website here?

File: docs/docsite/rst/collections_tech_preview.rst [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+
+Required Fields:
+    - ``namespace``: the namespace that the collection lives under. It must be a valid Python identifier,
+        may only contain alphanumeric characters and underscores. Additionally
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
        and may only contain alphanumeric characters and underscores. Additionally
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+        they cannot start with underscores or numbers and cannot contain consecutive
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
        the ``namespace`` cannot start with underscores or numbers and cannot contain consecutive
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 12]
```diff
@@ -0,0 +1,313 @@
...
+        underscores.
+    - ``name``: the collection's name. Has the same character restrictions as ``namespace``.
+    - ``version``: the collection's version. To upload to Galaxy, it must be compatible with semantic versioning.
+
+
+Optional Fields:
+    - ``dependencies``: A dictionary where keys are collections, and values are version
+      range 'specifiers <https://python-semanticversion.readthedocs.io/en/latest/#requirement-specification>`_,
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
      range 'specifiers <https://python-semanticversion.readthedocs.io/en/latest/#requirement-specification>`_.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 13]
```diff
@@ -0,0 +1,313 @@
...
+      it is good practice to depend on a version range to minimize conflicts, and pin to a
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
      It is good practice to depend on a version range to minimize conflicts, and pin to a
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 14]
```diff
@@ -0,0 +1,313 @@
...
+      major version to protect against breaking changes, ex: ``"user1.collection1": ">=1.2.2,<2.0.0"``
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
      a major version to protect against breaking changes. For example: ``"user1.collection1": ">=1.2.2,<2.0.0"``
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 15]
```diff
@@ -0,0 +1,313 @@
...
+      allow other collections as dependencies, not traditional roles.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
      This field allows  other collections as dependencies, not traditional roles.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 16]
```diff
@@ -0,0 +1,313 @@
...
+    - ``description``: A short summary description of the collection.
+    - ``license``: Either a single license or a list of licenses for content inside of a collection.
+      Galaxy currently only accepts `SPDX <https://spdx.org/licenses/>`_ licenses.
+    - ``tags``: a list of tags. These have the same character requirements as ``namespace`` and ``name``.
+    - ``repository``: URL of originating SCM repository.
+
+Docs
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
docs directory
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 17]
```diff
@@ -0,0 +1,313 @@
...
+-----
+
+Intended to keep general documentation for the collection here. Plugins and modules will still keep their specific documentaiton embeded, but here you can list the roles/plugins the colleciton provides, role requirements, user guides, etc. Currently we are looking at Markdown and the standard format for docs.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
Keep general documentation for the collection here. Plugins and modules will still keep their specific documentation embedded as Python docstrings. Use the ``docs`` folder to list the roles and plugins the collection provides, role requirements, user guides, and so on. Currently we are looking at Markdown as the standard format for documentation files, but this is subject to change.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 18]
```diff
@@ -0,0 +1,313 @@
...
+
+Updateing `ansible-doc <https://github.com/ansible/ansible/pull/57764>`_ to allow showing docs for plugins inside a collection::
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
We are `updating ansible-doc  <https://github.com/ansible/ansible/pull/57764>`_ to allow showing documentation for plugins inside a collection::
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 19]
```diff
@@ -0,0 +1,313 @@
...
+
+    ansible-doc -t lookup mycol.myname.lookup1
+
+It just requies FQCN to display specific plugins docs.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
The ``ansible-doc`` command requires the fully qualified collection name (FQCN) to display specific plugin documentation.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 20]
```diff
@@ -0,0 +1,313 @@
...
+
+
+Plugins
+-------
+
+ Under here a collection can host a 'per plugin type' specific directory, including ``module_utils`` which is usable not only by modules, but by any other plugin by using their 'fully qualified collection name' (FQCN for short). This is a way to distribute modules, lookups, filters, etc without having to import a role in every play.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
 Add a 'per plugin type' specific subdirectory here, including ``module_utils`` which is usable not only by modules, but by any other plugin by using their FQCN. This is a way to distribute modules, lookups, filters, and so on, without having to import a role in every play.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 21]
```diff
@@ -0,0 +1,313 @@
...
+Intended to keep general documentation for the collection here. Plugins and modules will still keep their specific documentaiton embeded, but here you can list the roles/plugins the colleciton provides, role requirements, user guides, etc. Currently we are looking at Markdown and the standard format for docs.
+
+Updateing `ansible-doc <https://github.com/ansible/ansible/pull/57764>`_ to allow showing docs for plugins inside a collection::
+
+    ansible-doc -t lookup mycol.myname.lookup1
+
+It just requies FQCN to display specific plugins docs.
+
+
+Plugins
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
plugins directory
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 22]
```diff
@@ -0,0 +1,313 @@
...
+-------
+
+ Under here a collection can host a 'per plugin type' specific directory, including ``module_utils`` which is usable not only by modules, but by any other plugin by using their 'fully qualified collection name' (FQCN for short). This is a way to distribute modules, lookups, filters, etc without having to import a role in every play.
+
+
+Roles
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
roles directory
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 23]
```diff
@@ -0,0 +1,313 @@
...
+      major version to protect against breaking changes, ex: ``"user1.collection1": ">=1.2.2,<2.0.0"``
+      allow other collections as dependencies, not traditional roles.
+    - ``description``: A short summary description of the collection.
+    - ``license``: Either a single license or a list of licenses for content inside of a collection.
+      Galaxy currently only accepts `SPDX <https://spdx.org/licenses/>`_ licenses.
+    - ``tags``: a list of tags. These have the same character requirements as ``namespace`` and ``name``.
+    - ``repository``: URL of originating SCM repository.
+
+Docs
+-----
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
---------------
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 24]
```diff
@@ -0,0 +1,313 @@
...
+
+Updateing `ansible-doc <https://github.com/ansible/ansible/pull/57764>`_ to allow showing docs for plugins inside a collection::
+
+    ansible-doc -t lookup mycol.myname.lookup1
+
+It just requies FQCN to display specific plugins docs.
+
+
+Plugins
+-------
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
------------------
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 25]
```diff
@@ -0,0 +1,313 @@
...
+
+ Under here a collection can host a 'per plugin type' specific directory, including ``module_utils`` which is usable not only by modules, but by any other plugin by using their 'fully qualified collection name' (FQCN for short). This is a way to distribute modules, lookups, filters, etc without having to import a role in every play.
+
+
+Roles
+-----
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
----------------
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 26]
```diff
@@ -0,0 +1,313 @@
...
+
+These are mostly the same as existing roles, but with a couple of limitations:
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
Collection roles are mostly the same as existing roles, but with a couple of limitations:
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 27]
```diff
@@ -0,0 +1,313 @@
...
+
+ - Role names are now limited to contain only lowercase alphanumeric characters, plus ``_`` and start with an alpha character.
+ - Roles cannot have their own plugins anymore, these must live in the collection and will be accessible to roles.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
 - Roles cannot have their own plugins any more. The plugins must live in the collection ``plugins`` directory and will be accessible to the collection roles.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 28]
```diff
@@ -0,0 +1,313 @@
...
+
+The directory name of the role is used as the role name, so therefore the directory name must comply with the
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
The directory name of the role is used as the role name. Therefore, the directory name must comply with the
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 29]
```diff
@@ -0,0 +1,313 @@
...
+above rules or not be usable.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
above role name rules.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 30]
```diff
@@ -0,0 +1,313 @@
...
+If a role name is encountered that does not match the above rules, the collection import into Galaxy will fail.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
The collection import into Galaxy will fail if a role name does not comply with these rules.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 31]
```diff
@@ -0,0 +1,313 @@
...
+
+You can migrate 'traditional roles' into a collection but they must follow the rules above, they will need to be renamed if they don't conform and any plugins will have to be moved or linked into the collection specific directories.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
You can migrate 'traditional roles' into a collection but they must follow the rules above. You man need to rename roles if they don't conform. You will have to move or link any role-based plugins to the collection specific directories.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 32]
```diff
@@ -0,0 +1,313 @@
...
+
+.. note::
+
+    For roles imported into Galaxy directly from a GitHub repository, setting the ``role_name`` value in the role's
+    metadata overrides the role name used by Galaxy. For collections, that value is ignored. When importing a
+    collection, Galaxy uses the role directory as the name of the role and ignores the ``role_name`` metadata value.
+
+Playbooks
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
playbooks directory
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 33]
```diff
@@ -0,0 +1,313 @@
...
+---------
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
--------------------
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 34]
```diff
@@ -0,0 +1,313 @@
...
+
+TBD.
+
+Tests
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
tests directory
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 35]
```diff
@@ -0,0 +1,313 @@
...
+-----
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
----------------
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 36]
```diff
@@ -0,0 +1,313 @@
...
+
+TBD. Expect tests for the collection itself, including Molecule files, to reside here.
+
+
+.. _creating_collections:
+
+Creating Collections
+====================
+
+Currently this is a work in progress, we created the `Mazer <https://galaxy.ansible.com/docs/mazer/index.html>`_ command line tool
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
This is currently is a work in progress. We created the `Mazer <https://galaxy.ansible.com/docs/mazer/index.html>`_ command line tool
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 37]
```diff
@@ -0,0 +1,313 @@
...
+available at the `Ansible Mazer project <https://github.com/ansible/mazer>`_. as a proof of concept for packaging,
+distributing and installing collections.  You can install via ``pip install mazer`` or checkout the code directly.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
distributing and installing collections.  You can install mazer with ``pip install mazer`` or checkout the code directly.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 38]
```diff
@@ -0,0 +1,313 @@
...
+
+.. Note::
+    All the docs below that use ``mazer`` might be updated to use another tool in the future as it has served it's purpose and it won't be updated any further.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
    All the documentation below that use ``mazer`` might be updated to use another tool in the future as ``mazer`` will not be updated in the future.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 39]
```diff
@@ -0,0 +1,313 @@
...
+
+We are working on integrating this into Ansible itself for 2.9, currently we have an `ansible-galaxy PR <https://github.com/ansible/ansible/pull/57106>`_ incorporating some of the commands into ``ansible-galaxy``. Currently it is not installable outside ansible, but we hope to land this into development soon so early adopters can test.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
We are working on integrating this into Ansible itself for 2.9. Currently we have an `ansible-galaxy PR <https://github.com/ansible/ansible/pull/57106>`_ incorporating some of the commands into ``ansible-galaxy``. Currently it is not installable outside ansible, but we hope to land this into development soon so early adopters can test.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 40]
```diff
@@ -0,0 +1,313 @@
...
+
+.. Note::
+    Any references to ``ansbile-galaxy`` below will be of a 'working version' either in this PR or subsequently in development, as such do it is subject to frequent change.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
    Any references to ``ansbile-galaxy`` below will be of a 'working version' either in this PR or subsequently in development. As such, the command and this documentation section is subject to frequent change.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 41]
```diff
@@ -0,0 +1,313 @@
...
+We also plan to update `Ansible Molecule <https://github.com/ansible/molecule>`_, for a full developer toolkit with integrated testing.
+
+In the end, to get started with authoring a new collection it should be as simple as:
+
+.. code-block:: bash
+
+    collecion_dir#>ansible-galaxy collection init
+
+
+And then populating the directories with the content you want inside the collection. For now you can clone from  https://github.com//bcoca/collection to get the directory structure if you want (or just create the directories as you need them).
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
And then populating the directories with the content you want inside the collection. For now you can optionally clone from  https://github.com//bcoca/collection to get the directory structure (or just create the directories as you need them).
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 42]
```diff
@@ -0,0 +1,313 @@
...
+    ├── releases
+    │   └── namespace_name-collection_name-1.0.12.tar.gz
+    └── ...
+
+.. note::
+        Changing the filename of the tarball in the release directory so that it doesn't match
+        the data in ``galaxy.yml`` will cause the import to fail.
+
+
+This tarball itself can be used to install the collection on target systems, it is mainly intended to upload to Galaxy as a distribution method, but you should be able to use directly.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
This tarball itself can be used to install the collection on target systems. It is mainly intended to upload to Galaxy as a distribution method, but you should be able to use directly.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 43]
```diff
@@ -0,0 +1,313 @@
...
+
+Publishing Collections
+======================
+
+We are in the process of updating Ansible Galaxy to allow for it to manage collections as it has been managing roles until now.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
We are in the process of updating Ansible Galaxy to manage collections as it manages roles until now.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 44]
```diff
@@ -0,0 +1,313 @@
...
+
+
+Upload From the Galaxy Website
+------------------------------
+
+Go to the `My Content </my-content/namespaces>`_ page, and click the *Add Content* button on one of your namespaces. From
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
Go to the `My Content </my-content/namespaces>`_ page, and click the **Add Content** button on one of your namespaces. From
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 45]
```diff
@@ -0,0 +1,313 @@
...
+the *Add Content* dialogue, click *Upload New Collection*, and select the collection archive file from your local
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
the **Add Content** dialogue, click **Upload New Collection**, and select the collection archive file from your local
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 46]
```diff
@@ -0,0 +1,313 @@
...
+filesystem.
+
+When uploading collections it doesn't actually matter which namespace you select. The collection will be uploaded to the
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
When uploading collections it doesn't  matter which namespace you select. The collection will be uploaded to the
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 47]
```diff
@@ -0,0 +1,313 @@
...
+namespace specified in the collection metadata specified in the ``galaxy.yml`` file. If you're not an owner of the
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
namespace specified in the collection metadata  in the ``galaxy.yml`` file. If you're not an owner of the
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 48]
```diff
@@ -0,0 +1,313 @@
...
+namespace, the upload request will fail.
+
+Once a collection has been uploaded and accepted by Galaxy, you will be redirected to the My Imports page, displaying output from the
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
Once Galaxy uploads and accepts a collection, you will be redirected to the **My Imports** page, which displays output from the
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 49]
```diff
@@ -0,0 +1,313 @@
...
+import process, including any errors or warnings about the metadata and content contained in the collection.
+
+Upload Using Mazer
+------------------
+
+Collection artifacts can be uploaded with Mazer, as shown in the following example:
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
You can upload ollection artefacts with Mazer, as shown in the following example:
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 50]
```diff
@@ -0,0 +1,313 @@
...
+
+.. code-block:: bash
+
+    mazer publish --api-key=SECRET path/to/namespace_name-collection_name-1.0.12.tar.gz
+
+The above will trigger an import process, just as if the collection had been uploaded through the Galaxy website. Use the My Imports
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
The above command triggers an import process, just as if the collection had been uploaded through the Galaxy website. Use the **My Imports**
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 51]
```diff
@@ -0,0 +1,313 @@
...
+page to view the output from the import process.
+
+Your API key can be found on `the preferences page in Galaxy </me/preferences>`_.
+
+To learn more about Mazer, view :doc:`../mazer/index`.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
To learn more about Mazer, see `Mazer <https://galaxy-dev.ansible.com/docs/mazer/index.html>`.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 52]
```diff
@@ -0,0 +1,313 @@
...
+
+
+Collection Versions
+-------------------
+
+Once a version of a collection has been uploaded it cannot be deleted or modified, so make sure that everything looks okay before
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
Once you upload a version of a collection, you cannot delete or modify that version. Ensure that everything looks okay before
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 53]
```diff
@@ -0,0 +1,313 @@
...
+uploading. The only way to change a collection is to release a new version. The latest version of a collection (by highest version number)
+will be the version displayed everywhere in Galaxy; however, users will still be able to download older versions.
+
+
+Installing Collections
+======================
+
+There are several ways, the most common will be:
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
The recommended way to install a collection is:
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 54]
```diff
@@ -0,0 +1,313 @@
...
+assuming the collection is hosted in Galaxy.
+
+You can also use a tarball resulting from your build:
+
+.. code-block:: bash
+
+   #> ansible-galaxy install  mynamespace.mycollection.0.1.0.tgz -p /path
+
+
+As a path you should use one of the values configured in `COLLECTINS_PATHS <https://docs.ansible.com/ansible/latest/reference_appendices/config.html#collections-paths>`_. This is also where Ansible itself will expect to find them when attempting to use a collection.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
As a path you should use one of the values configured in `COLLECTINS_PATHS <https://docs.ansible.com/ansible/latest/reference_appendices/config.html#collections-paths>`_. This is also where Ansible itself will expect to find collections when attempting to use them.
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 55]
```diff
@@ -0,0 +1,313 @@
...
+Once installed, you can reference collection content by its FQCN:
+
+.. code-block:: yaml
+
+     - hosts: all
+       tasks:
+         - myname.mycol.mymodule:
+             option1: value
+
+This works for roles or any type of plugin distributed:
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
This works for roles or any type of plugin distributed within the collection:
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 56]
```diff
@@ -0,0 +1,313 @@
...
+       collections:
+        - myname.mycol
+       tasks:
+         - include_role:
+             name: role1
+
+         - debug:
+             msg: '{{ lookup("lookup1", 'param1')|filter1 }}'
+
+This keyword creates a 'search path' for non namespaced plugin refreences, it does not import roles or anything else.
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst]:
```suggestion
This keyword creates a 'search path' for non namespaced plugin references. It does not import roles or anything else.
```

**Review by samccann — COMMENTED** [review]
Documentation nit police comments :-)  Thanks for writing all this!

File: docs/docsite/rst/collections_tech_preview.rst [hunk 57]
```diff
@@ -0,0 +1,313 @@
+.. _collections:
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst:3]:
```suggestion
:orphan:

.. _collections:
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 58]
```diff
@@ -0,0 +1,315 @@
+:orphan:
+
+.. _collections:
+
+Collections
+===========
+
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst:7]:
```suggestion

.. contents::
   :local:
```

File: docs/docsite/rst/collections_tech_preview.rst [hunk 59]
```diff
@@ -0,0 +1,317 @@
...
+.. contents::
+   :local:
```
Comment (samccann) [inline docs/docsite/rst/collections_tech_preview.rst:9]:
```suggestion
   :local:
   
```

**Comment by jborean93** [comment]
@bcoca we need to revisit the structure of `galaxy.yml` as set in this PR. Mazer added 2 fields `readme`, and an optional `license_file` that is used when generating the MANIFEST.json in the build artifact.

**Comment by samccann** [comment]
@jborean93 do you have a pointer to these two new fields and any further details about what a user does (or doesn't) do with them?

**Comment by bcoca** [comment]
@jborean93, i removed since we had discussed they were not useful during one of the meetings, i would ping @thaumos to confirm the 'real state' of the spec

**Comment by samccann** [comment]
Document is now publicly available at:
https://docs.ansible.com/ansible/devel/collections_tech_preview.html

**Pull request merged**